gloo/docs/user_guides/tls_setup.md

```diff
@@ -0,0 +1,317 @@
+--ssl-sni-domains animalstore.example.com
+```
+
+{{% notice note %}}
+As you can see in the previous step, we need to specify the SNI domains that will match for this certificate with the `--ssl-sni-domains` parameter. If you do NOT specify this parameter, Envoy will become confused about which certificates to serve because there will effectively be two (or more) with no qualifying information. If that's the case, you can expect to see logs similar to the following in your `gateway-proxy` logs:
+
+{{< highlight bash >}}
+gateway-proxy-9b55c99c7-x7r7c gateway-proxy [2019-03-20 19:01:01.763][6][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:70] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener listener-::-8443: error adding listener '[::]:8443': multiple filter chains with the same matching rules are defined
+{{< /highlight >}}  
+
+If you end up with logs like that, double check your SNI settings.
+
+{{% /notice %}}
+
+
+Lastly, let's add a route for this VirtualService:
+
+```bash
+glooctl add route --name animal\
+    --path-exact /animals \
+    --dest-name default-petstore-8080 \
+    --prefix-rewrite /api/pets
+```
+
+Note, we're giving this service a different API, namely `/animals` instead of `/sample-route-1`.
+
+Now if we get the VirtualService, we should see this one set up with a different cert/secret:
+
+```bash
+glooctl get virtualservice animal -o yaml
+```
+
+{{< highlight yaml "hl_lines=2-5" >}}
+---
+displayName: animal
+sslConfig:
+  secretRef:
+    name: animal-certs
+    namespace: gloo-system
+status:
+  reportedBy: gateway
+  state: Accepted
+virtualHost:
+  domains:
+  - '*'
+  routes:
+  - matcher:
+      exact: /animals
+    routeAction:
+      single:
+        upstream:
+          name: default-petstore-8080
+          namespace: gloo-system
+    routePlugins:
+      prefixRewrite:
+        prefixRewrite: /api/pets
+{{< /highlight >}}     
+
+If everything up to this point looks good, let's try to query the service and make sure to pass in the qualifying `Host` information so that Envoy can serve the correct certificates.
+
+
+```bash
+curl -k -H "Host: animalstore.example.com"  \
+$(glooctl proxy url --port https)/animals
+
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+
+
+### Understanding how it all works
+By default, when a VirtualService does NOT have any SSL/TLS configuration, it will be attached to the HTTP listener that we have for Gloo proxy (listening on port `8080` by default, but exposed in Kubernetes on port `80` in the `gateway-proxy` service). When we add the SSL/TLS configuration, that VirtualService will automatically become bound to the HTTPS port (listening on port `8443` on the gateway-proxy, but mapped to port `443` on the Kubernetes service. 
```

Comment:
Are there cases where user would want to have different defaults either within cluster (8080/8443) or externally (80/443)? If so, is there a way to change defaults?

Reply:
Yah I'm going to write another doc on exposing custom listeners/changing the defaults and will link from this doc